python/cudf/cudf/_lib/aggregation.pxd

```diff
@@ -4,7 +4,9 @@ from libcpp.memory cimport unique_ptr
 from cudf._lib.cpp.aggregation cimport aggregation
 
 
-cdef unique_ptr[aggregation] make_aggregation(op, kwargs=*) except *
-
 cdef class Aggregation:
     cdef unique_ptr[aggregation] c_obj
+
+# TODO: the old version of this function was declared as except *, I'm not sure
+# why but I should figure that out before calling this done.
```

Comment:
Any `cdef` functions that return a non-Python object have no way of communicating exceptions to their callers unless you mark them `except *` or `except ?` (the latter when you can specify a specific return value).

It's a good idea to do this even if the underlying C++ code is "guaranteed" not to raise any exceptions because Python can very easily sneak into the implementation, or there may be other errors like `RecursionError` that can originate from inside the function.

Reply:
Thanks for finding, I looked into the syntax but forgot to revisit. I was running into some issues with this syntax because of the value of the return type, but perhaps at this stage I've changed enough that it will just work.

Reply:
I don't think we need to do `except *` anymore, now that we're returning a Python object. Just commented as an FYI :)

Reply:
Ah. The reason this gave errors is that Cython classes are Python types, in the sense that they are still represented by `PyObject *` pointers, which means that error propagation can proceed via the normal route in CPython: if a function returns NULL, the calling code needs to check for exceptions. Before, we were actually returning a true C(++) object, so this wasn't the case. We don't need this anymore.

Reply:
I wish threads would update without me refreshing the page :) thanks for confirming